playbooks/secrets.yml

```diff
@@ -0,0 +1,85 @@
+---
+- hosts: all:!external
+  gather_facts: false
+  tasks:
+    - name: Ensure no secrets are still provided via group_vars and/or hostvars
+      assert:
+        that:
+          - repo_db_password is not defined
+          - sync_db_password is not defined
+          - reposearch_shared_secret is not defined
+        msg:
+          "The current playbook version expect secrets to be provided in vars/secrets.yml.
+          If you are upgrading from a previous playbook version, please read to docs/playbook-upgrade.md"
+        quiet: true
```

Comment:
main playbook should stop if any host has such an env.
Here I think if one var is present on a single host only that host will be taken out of the play but play will go on.

Reply:
reproduced creating a `group_vars/database.yml` with a credential in it, `any_errors_fatal` helps